src/main/scala/com/microsoft/hyperspace/index/dataskipping/util/DataSkippingFileIndex.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (2021) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.dataskipping.util
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.execution.datasources._
+import org.apache.spark.sql.functions.isnull
+import org.apache.spark.sql.types.StructType
+
+import com.microsoft.hyperspace.index.{FileIdTracker, IndexConstants}
+
+class DataSkippingFileIndex(
```

Comment:
Can we move this to `src/main/scala/com/microsoft/hyperspace/index/execution/datasources` for consistency? or at least `src/main/scala/com/microsoft/hyperspace/index/dataskipping/execution/datasources`? (considering the location of FileIndex.scala: https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileIndex.scala

This class looks a kind of plan rather than util

And could you add some comment for the class?

Reply:
Moved to `execution` as I don't expect we'll have many files for this directory.